lib/bundler/friendly_errors.rb

```diff
@@ -23,13 +23,12 @@ def log_error(error)
         Bundler.ui.error error.message
       when LoadError
         raise error unless error.message =~ /cannot load such file -- openssl|openssl.so|libcrypto.so/
-        Bundler.ui.error "\nCould not load OpenSSL."
+        Bundler.ui.error "\nCould not load OpenSSL. #{error.class}: #{error}\n#{error.backtrace.join("\n  ")}"
         Bundler.ui.warn <<-WARN, :wrap => true
           You must recompile Ruby with OpenSSL support or change the sources in your \
           Gemfile from 'https' to 'http'. Instructions for compiling with OpenSSL \
           using RVM are available at https://rvm.io/packages/openssl.
         WARN
```

Comment:
Since we are here, do you think this message adds much value? I think I would remove this message entirely or maybe change the message to:

> Could not load OpenSSL, you ruby might not be compiled with OpenSSL support. The underlying error was #{error.class}: #{error}\n#{error.backtrace.join("\n  ")}"

The reason is:

* I don't think we want to explicitly recommend switching source the http.
* The RVM advice seems to specific to me, and it redirects to a page that's not very clear about how it helps to fix this issue.

What do you think? If you think this is too unrelated to this PR, I can tackle this myself separately.

Reply:
I totally agree that this message isn't helpful. I'm happy to get you a new PR updating this message tomorrow morning if you want to merge this one now. :)

Reply:
Sure!